src/OrleansAzureUtils/Providers/Lease/AzureBlobLeaseProvider.cs

```diff
@@ -0,0 +1,140 @@
+        }
+
+        private async Task<AcquireLeaseResult> Acquire(string category, LeaseRequest leaseRequest)
+        {
+            try
+            {
+                var blob = this.container.GetBlockBlobReference(GetBlobName(category, leaseRequest.ResourceKey));
+                blob.Properties.ContentType = "application/json";
+                //create this blob
+                await blob.UploadTextAsync("blob");
+                var leaseId = await blob.AcquireLeaseAsync(leaseRequest.Duration);
+                return new AcquireLeaseResult(new AcquiredLease(leaseRequest.ResourceKey, leaseRequest.Duration, leaseId, DateTime.UtcNow), ResponseCode.OK, null);
+            }
+            catch (StorageException e)
+            {
+                ResponseCode statusCode;
+                //This mapping is based on references : https://docs.microsoft.com/en-us/rest/api/storageservices/blob-service-error-codes
+                // https://docs.microsoft.com/en-us/rest/api/storageservices/Lease-Blob?redirectedfrom=MSDN
+                switch (e.RequestInformation.HttpStatusCode)
+                {
+                    case 404:
+                    case 409: 
+                    case 412: statusCode = ResponseCode.LeaseNotAvailable; break;
+                    default: statusCode = ResponseCode.TransientFailure; break;
+                }
+                return new AcquireLeaseResult(null, statusCode, e);
+            }
+        }
+
+        public async Task Release(string category, AcquiredLease[] aquiredLeases)
+        {
+            await InitContainerIfNotExistsAsync();
+            var tasks = new List<Task>();
+            foreach (var acquiredLease in aquiredLeases)
+            {
+                tasks.Add(Release(category, acquiredLease));
+            }
+            await Task.WhenAll(tasks);
+        }
+
+        private Task Release(string category, AcquiredLease acquiredLease)
+        {
+            var blob = this.container.GetBlobReference(GetBlobName(category, acquiredLease.ResourceKey));
+            return blob.ReleaseLeaseAsync(AccessCondition.GenerateLeaseCondition(acquiredLease.Token));
+        }
+
+        public async Task<AcquireLeaseResult[]> Renew(string category, AcquiredLease[] aquiredLeases)
+        {
+            await InitContainerIfNotExistsAsync();
+            var tasks = new List<Task<AcquireLeaseResult>>();
+            foreach (var acquiredLease in aquiredLeases)
+            {
+                tasks.Add(Renew(category, acquiredLease));
+            }
+            //Task.WhenAll will return results for each task in an array, in the same order of supplied tasks
+            return await Task.WhenAll(tasks);
+        }
+
+        private async Task<AcquireLeaseResult> Renew(string category, AcquiredLease acquiredLease)
+        {
+            var blob = this.container.GetBlobReference(GetBlobName(category, acquiredLease.ResourceKey));
```

Comment:
Maybe this is pre-mature optimization, but you could cache the blobName on a property in a prviate subclass of `AcquiredLease`. You could go all the way and cache the blob ref.